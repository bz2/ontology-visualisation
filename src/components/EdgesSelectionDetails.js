import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import actions from '../store/actions'
import EdgeSelectionDetailsRow from './EdgeSelectionDetailsRow'

const EdgesSelectionDetails = ({
  edgeId,
  objectPropertiesFromApi,
  edgesConnections,
}) => {
  const { t } = useTranslation()

  const selectedEdge = objectPropertiesFromApi[edgeId]

  const tableRowNames = Object.keys(selectedEdge).filter((key) => typeof selectedEdge[key] !== 'object'
    && !key.includes('label')).sort()

  let connections = []

  if (edgesConnections[edgeId]) {
    connections = edgesConnections[edgeId]
  }

  return (
    <div className="edges-selection-details">
      <div className="edges-selection-details-title">
        {t('properties')}
      </div>

      <div className="edges-selection-details-table">
        <table>
          <thead>
            <tr>
              <th />
              <th />
            </tr>
          </thead>
          <tbody>
            {
              tableRowNames.map((tableRowName) => (
                <tr
                  key={`details-row-${tableRowName}`}
                >
                  <td>
                    {tableRowName}
                  </td>
                  <td>
                    {selectedEdge[tableRowName] || t('null')}
                  </td>
                </tr>
              ))
            }
          </tbody>
        </table>
      </div>

      <div className="edges-selection-details-title">
        {t('nodesProperties')}
      </div>

      {
        connections.length > 0 ? (
          <div className="edges-selection-details-table">
            <table>
              <thead>
                <tr>
                  <th />
                  <th>{t('from')}</th>
                  <th>{t('to')}</th>
                </tr>
              </thead>
              <tbody>
                {
                  connections.map((connection) => (
                    <EdgeSelectionDetailsRow
                      key={`edge-relationship-row-${connection.from}-${connection.to}`}
                      connection={connection}
                    />
                  ))
                }
              </tbody>
            </table>
          </div>
        ) : (
          <>
            {t('noRelationships')}
          </>
        )
      }

    </div>
  )
}

EdgesSelectionDetails.propTypes = {
  edgeId: PropTypes.string.isRequired,
  objectPropertiesFromApi: PropTypes.shape().isRequired,
  edgesConnections: PropTypes.shape().isRequired,
}

const mapToProps = ({
  objectPropertiesFromApi,
  edgesConnections
}) => ({
  objectPropertiesFromApi,
  edgesConnections
})

export default connect(
  mapToProps,
  actions
)(EdgesSelectionDetails)
