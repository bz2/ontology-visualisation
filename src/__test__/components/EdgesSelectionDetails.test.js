import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EdgesSelectionDetails from '../../components/EdgesSelectionDetails'
import { OwlObjectProperties } from '../fixtures/test-ontology-object-properties'

const setup = () => {
  const props = {
    edgesConnections: [{
      from: 'http://webprotege.stanford.edu/R9BdIrtS5xNdcAPHLf4JaEE',
      predicate: 'http://webprotege.stanford.edu/R15RMwxh0pmeZADFPUrcpM',
      to: 'http://webprotege.stanford.edu/Rf8re1EdmLU47EIlLtCpzx'
    }],
    objectPropertiesFromApi: OwlObjectProperties,
    edgeId: 'http://webprotege.stanford.edu/R15RMwxh0pmeZADFPUrcpM'
  }

  const component = shallow(<EdgesSelectionDetails {...props} />)

  return {
    component,
    props
  }
}

describe('EdgesSelectionDetails', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot ', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
