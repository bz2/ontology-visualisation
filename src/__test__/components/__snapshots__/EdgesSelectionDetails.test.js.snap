// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EdgesSelectionDetails should match snapshot  1`] = `
<Connect
  actions={
    Object {
      "addNumber": [Function],
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
    }
  }
  edgeId="http://webprotege.stanford.edu/R15RMwxh0pmeZADFPUrcpM"
  mapToProps={null}
>
  <Component />
</Connect>
`;
